### PR TITLE
Load books to vuex

### PR DESCRIPTION
**Pull request by GiulianoScaglioni**
## Summary
- Books module for vuex
- Integrate dashboard with vuex

## Screenshot
demo-vuex

## Trello card
[https://trello.com/c/1DxqkhBU/19-vuex](https://trello.com/c/1DxqkhBU/19-vuex)

### CODE REVIEW

File: training-vue/wolox-books/src/views/Dashboard.vue [hunk 1]
```diff
@@ -8,23 +8,20 @@
...
-  },
   mounted () {
-    bookService.list().then(response => {
-      this.books = response.data
-    })
+    this.$store.dispatch('books/load')
+  },
+  computed: {
+    books () {
+      return this.$store.getters['books/list']
```
Comment (mcallegari10) [inline training-vue/wolox-books/src/views/Dashboard.vue]: Instead, use `mapGetters` function from Vuex https://vuex.vuejs.org/guide/getters.html

File: training-vue/wolox-books/src/views/Dashboard.vue [hunk 2]
```diff
@@ -8,23 +8,20 @@
...
-  data () {
-    return {
-      books: []
-    }
-  },
   mounted () {
-    bookService.list().then(response => {
-      this.books = response.data
-    })
+    this.$store.dispatch('books/load')
```
Comment (mcallegari10) [inline training-vue/wolox-books/src/views/Dashboard.vue]: Use `mapActions` from Vuex

File: training-vue/wolox-books/src/vuex/books.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  actions: {
+    load (context) {
+      bookService.list().then(response => {
+        context.commit('setList', { list: response.data })
+      })
+    }
+  },
+  getters: {
+    list (state) {
+      return state.list
```
Comment (mcallegari10) [inline training-vue/wolox-books/src/vuex/books.js]: Better use a one-line arrow function :stuck_out_tongue_closed_eyes:

File: training-vue/wolox-books/src/vuex/books.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { bookService } from '@/services/book'
+
+export default {
+  namespaced: true,
+  state: {
+    list: []
+  },
+  mutations: {
+    setList (state, payload) {
```
Comment (mcallegari10) [inline training-vue/wolox-books/src/vuex/books.js]: Instead of payload, you can use destructuring to take the `list` :smile:

**Review by mcallegari10 — CHANGES_REQUESTED** [review]

**Review by FrankIglesias — APPROVED** [review]

**Pull request merged**